tools/csv-gen/csv-gen.go

```diff
@@ -0,0 +1,690 @@
+				},
+			},
+		},
+	}
+
+	opMajor, opMinor, opMicro := activemqartemis.MajorMinorMicro(version.Version)
+	csvFile := "deploy/olm-catalog/" + csv.CsvDir + "/" + opMajor + "." + opMinor + "." + opMicro + "/" + csvVersionedName + ".clusterserviceversion.yaml"
+
+	imageName, _, _ := activemqartemis.GetImage(deployment.Spec.Template.Spec.Containers[0].Image)
+	relatedImages := []image{}
+
+	if csv.OperatorName == "activemq-artemis-operator" {
+		templateStruct.Annotations["certified"] = "false"
+		deployFile := "deploy/operator.yaml"
+		createFile(deployFile, deployment)
+		roleFile := "deploy/role.yaml"
+		createFile(roleFile, role)
+	}
+	relatedImages = append(relatedImages, image{Name: imageName, Image: deployment.Spec.Template.Spec.Containers[0].Image})
+
+	imageRef := constants.ImageRef{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: oimagev1.SchemeGroupVersion.String(),
+			Kind:       "ImageStream",
+		},
+		Spec: constants.ImageRefSpec{
+			Tags: []constants.ImageRefTag{
+				{
+					// Needs to match the component name for upstream and downstream.
+					Name: "amq7/amq-broker-rhel8-operator",
+					From: &corev1.ObjectReference{
+						// Needs to match the image that is in your CSV that you want to replace.
+						Name: deployment.Spec.Template.Spec.Containers[0].Image,
+						Kind: "DockerImage",
+					},
+				},
+			},
+		},
+	}
+
+	sort.Sort(sort.Reverse(sort.StringSlice(activemqartemis.SupportedVersions)))
+
+	imageRef.Spec.Tags = append(imageRef.Spec.Tags, constants.ImageRefTag{
+		Name: constants.Broker75Component,
+		From: &corev1.ObjectReference{
+			Name: constants.Broker75ImageURL,
+			Kind: "DockerImage",
+		},
+	})
+
+	relatedImages = append(relatedImages, getRelatedImage(constants.Broker75ImageURL))
+
+	if GetBoolEnv("DIGESTS") {
+
+		for _, tagRef := range imageRef.Spec.Tags {
+
+			if _, ok := imageShaMap[tagRef.From.Name]; !ok {
+				imageShaMap[tagRef.From.Name] = ""
+				imageName, imageTag, imageContext := activemqartemis.GetImage(tagRef.From.Name)
+				repo := imageContext + "/" + imageName
+
+				if strings.Contains(tagRef.From.Name, "quay.io") {
+					digests, err := util.RetriveFromQuayIO(repo, imageTag)
+					if err != nil {
+						fmt.Fprintln(os.Stderr, err)
+					}
+					if len(digests) > 0 {
+						imageShaMap[tagRef.From.Name] = strings.ReplaceAll(tagRef.From.Name, ":"+imageTag, "@"+digests[0])
+					} else {
+						fmt.Printf("could not get the digest for tag %s from registry %s \n", imageTag, tagRef.From.Name)
+					}
+
+				} else {
+
+					digests, err := util.RetriveFromRedHatIO(repo, imageTag)
+					if err != nil {
+						fmt.Fprintln(os.Stderr, err)
+					}
+					if len(digests) > 1 {
+						imageShaMap[tagRef.From.Name] = strings.ReplaceAll(tagRef.From.Name, ":"+imageTag, "@"+digests[len(digests)-1])
+					}
+
+				}
+			}
+		}
+	}
+
+	//imageFile := "deploy/olm-catalog/" + csv.CsvDir + "/" + opMajor + "." + opMinor + "." + opMicro + "/" + "image-references"
+	//createFile(imageFile, imageRef)
```

Comment:
is this a useful comment to keep?

Reply:
This required in case if we need image-references in the future. added comments for more describing inside the code.